### PR TITLE
Extend ngraph for AdaptivePool operations

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Add AdaptiveMaxPool-8 and AdaptiveAvgPool-8 to ngraph*

### Tickets:
 - *56360*

### CODE REVIEW

File: ngraph/core/include/ngraph/op/adaptive_avg_pool.hpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                ///
+                /// \param    output_shape    1D tensor describing output shape for spatial
+                ///                           dimensions.
+                ///
+                AdaptiveAvgPool(const Output<Node>& data, const Output<Node>& output_shape);
+
+                void validate_and_infer_types() override;
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                virtual std::shared_ptr<Node>
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/adaptive_avg_pool.hpp:38]: nitpitck: `virtual` is redundant

File: ngraph/test/type_prop/adaptive_avg_pool.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    auto data = make_shared<op::Parameter>(element::f32, arg_shape);
+    auto out_shape = op::Constant::create<int64_t>(element::i64, Shape{2}, output_shape);
+    auto adaptive_pool = make_shared<op::v8::AdaptiveAvgPool>(data, out_shape);
+
+    ASSERT_TRUE(adaptive_pool->get_output_partial_shape(0).same_scheme({1, 6, 5, 7}));
+}
+
+TEST(type_prop, adaptive_avg_pool_dyn_batch)
+{
+    const PartialShape arg_shape{Dimension::dynamic(), 6, 8, 9};
```
Comment (popovaan) [inline ngraph/test/type_prop/adaptive_avg_pool.cpp:26]: Maybe test with dynamic rank can be added?
Something like:
const PartialShape arg_shape = PartialShape::dynamic()?

**Review by popovaan — APPROVED** [review]

File: ngraph/test/type_prop/adaptive_max_pool.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    const PartialShape arg_shape{1, 6};
+    const vector<int64_t> output_shape{1};
+
+    auto data = make_shared<op::Parameter>(element::f32, arg_shape);
+    auto out_shape = op::Constant::create<int64_t>(element::i64, Shape{}, output_shape);
+
+    EXPECT_THROW(make_shared<op::v8::AdaptiveMaxPool>(data, out_shape), NodeValidationFailure);
+}
+
+TEST(type_prop, adaptive_max_pool_wrong_out_shape)
```
Comment (ilyachur) [inline ngraph/test/type_prop/adaptive_max_pool.cpp:119]: Could you please add tests for `visit_attributes`?

**Review by sadolini — APPROVED** [review]

File: ngraph/test/visitors/op/adaptive_avg_pool.cpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+TEST(attributes, adaptive_avg_pool_op)
+{
+    NodeBuilder::get_ops().register_factory<opset8::AdaptiveAvgPool>();
+    const auto A = make_shared<op::Parameter>(element::f32, Shape{1, 3, 5, 4});
+    const auto out_shape = op::Constant::create<int64_t>(element::i64, Shape{2}, {4, 3});
+
+    const auto adaptive_pool = make_shared<opset8::AdaptiveAvgPool>(A, out_shape);
+    NodeBuilder builder(adaptive_pool);
+
+    const auto expected_attr_count = 0;
```
Comment (ilyachur) [inline ngraph/test/visitors/op/adaptive_avg_pool.cpp:25]: Could you reuse approach from this PR: https://github.com/openvinotoolkit/openvino/pull/6181

File: ngraph/test/visitors/op/adaptive_max_pool.cpp [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+{
+    NodeBuilder::get_ops().register_factory<opset8::AdaptiveMaxPool>();
+    const auto A = make_shared<op::Parameter>(element::f32, Shape{1, 3, 5, 4});
+    const auto out_shape = op::Constant::create<int64_t>(element::i64, Shape{2}, {4, 3});
+
+    const auto adaptive_pool = make_shared<opset8::AdaptiveMaxPool>(A, out_shape);
+    NodeBuilder builder(adaptive_pool);
+
+    const auto expected_attr_count = 1;
+    EXPECT_EQ(builder.get_value_map_size(), expected_attr_count);
```
Comment (ilyachur) [inline ngraph/test/visitors/op/adaptive_max_pool.cpp:26]: I think you also need to check value in the map

**Review by ilyachur — COMMENTED** [review]
Why did you update oneDNN?

File: ngraph/test/visitors/op/adaptive_avg_pool.cpp (hunk 4 above)
Comment (mvafin) [reply ngraph/test/visitors/op/adaptive_avg_pool.cpp:25]: How? It is not unary op

**Comment by mvafin** [comment]
> Why did you update oneDNN?

I didn't and I don't see that in diff

File: ngraph/test/visitors/op/adaptive_avg_pool.cpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/test/visitors/op/adaptive_avg_pool.cpp:25]: Please take a look to PR. This PR contains `visitor_without_atrribute` test.

**Comment by ilyachur** [comment]
> > Why did you update oneDNN?
> 
> I didn't and I don't see that in diff

Looks like the incorrect git merge:
Screenshot 2021-06-22 at 09 30 09

File: ngraph/test/visitors/op/adaptive_avg_pool.cpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/test/visitors/op/adaptive_avg_pool.cpp:25]: @mvafin You was right my error. We cannot create AdaptiveAvgPool

**Review by ilyachur — APPROVED** [review]

**Pull request merged**